src/process_utility.c

```diff
@@ -1191,12 +1197,105 @@ process_cluster_start(Node *parsetree, ProcessUtilityContext context)
 		/* Start a new transaction for the cleanup work. */
 		StartTransactionCommand();
 
+		/* Mark the hypertable main table as clustered. */
+		chunk_index_mark_clustered(ht->main_table_relid, index_relid);
+
 		/* Clean up working storage */
 		MemoryContextDelete(mcxt);
+		reclustered = true;
 	}
 
+recluster_cleanup:
 	cache_release(hcache);
-	return false;
+	return reclustered;
+}
+
+static bool
+recluster_chunk(ClusterStmt *stmt, Cache *hcache, bool is_top_level)
+{
+	Oid			index_relid;
+	Chunk	   *chunk;
+	Hypertable *ht;
+	ChunkIndexMapping *cim;
+	Oid			chunk_id = RangeVarGetRelid(stmt->relation, NoLock, true);
+
+	if (!OidIsValid(chunk_id))
+		return false;
+
+	chunk = chunk_get_by_relid(chunk_id, 0, false);
+
+	/* If the relation wasn't a chunk let postgres's cluster handle it. */
+	if (NULL == chunk)
+		return false;
+
+	ht = hypertable_cache_get_entry(hcache, chunk->hypertable_relid);
+
+	recluster_permission_check(ht, is_top_level);
+
+	/* first check the hypertable for the index, also contains the NULL check */
```

Comment:
I actually think the NULL case should be handled differently. If we get NULL we should find_clustered_index on the chunk not the hypertable. A hypertable cluster will mark the chunk index as the clustered index in `chunk_index_mark_clustered(cim->chunkoid, cim->indexoid);` right?